include/swift/Reflection/TypeRefBuilder.h

```diff
@@ -558,9 +558,7 @@ class TypeRefBuilder {
     auto kind = node->getKind();
     // Kinds who have a "BoundGeneric..." variant.
     if (kind != Node::Kind::Class && kind != Node::Kind::Structure &&
-        kind != Node::Kind::Enum && kind != Node::Kind::Protocol &&
-        kind != Node::Kind::OtherNominalType && kind != Node::Kind::TypeAlias &&
-        kind != Node::Kind::Function)
```

Comment:
@augusto2112 you added these non-{class,struct,enum} types originally, were they needed originally but not now?

Reply:
No, that was a bug. We found that the demangler only deals with classes structs and enums when building a detangle tree from a typeref https://github.com/apple/swift/blob/main/stdlib/public/runtime/Demangle.cpp#L153-L157